src/bundle/Controller/ContentViewController.php

```diff
@@ -59,6 +59,29 @@ public function detailsTabAction(ContentView $view)
         return $view;
     }
 
+    public function versionsTabAction(ContentView $view)
+    {
+        $contentInfo = $view->getContent()->getVersionInfo()->getContentInfo();
+        $contentService = $this->getRepository()->getContentService();
+        $versions = $contentService->loadVersions($contentInfo);
+
+        $versionFilter = $this->container->get('ezsystems.platform_ui.hybrid.filter.version_filter');
```

Comment:
It's done like that in one controller, but several others do not. Good confirmation bias example ;)

Jokes aside, we may wanna try to auto-wiring of controller action arguments, and have the container pass the Filter service to the action method instead: https://symfony.com/doc/current/controller.html#controller-accessing-services.

Could you give it a try @maxbaldanza ?